android/app/src/main/java/com/mulberry/ody/presentation/notification/NotificationHelper.kt

```diff
@@ -2,62 +2,84 @@ package com.mulberry.ody.presentation.notification
 
 class NotificationHelper(private val context: Context) {
     init {
-        setNotificationChannel()
+        createNotificationChannel()
     }
 
-    private fun makeNotificationMessage(
+    fun showNotification(
         type: NotificationType,
         nickname: String,
+        meetingId: String,
+        meetingName: String,
+    ) {
+        val message = getNotificationMessage(type, nickname, meetingName)
+        val pendingIntent = getPendingIntent(type, meetingId)
+        showNotification(message, pendingIntent)
+    }
+
+    private fun getNotificationMessage(
+        type: NotificationType,
+        nickname: String,
+        meetingName: String,
     ): String =
         when (type) {
-            NotificationType.ENTRY -> {
-                context.getString(R.string.fcm_notification_entry, nickname)
-            }
-
-            NotificationType.DEPARTURE_REMINDER -> {
-                context.getString(R.string.fcm_notification_departure_reminder, nickname)
-            }
-
-            NotificationType.DEPARTURE -> {
-                context.getString(R.string.fcm_notification_departure, nickname)
-            }
+            NotificationType.ENTRY -> context.getString(R.string.fcm_notification_entry, nickname)
+            NotificationType.DEPARTURE_REMINDER -> context.getString(R.string.fcm_notification_departure_reminder, nickname)
+            NotificationType.NUDGE -> context.getString(R.string.fcm_notification_nudge, nickname)
+            NotificationType.MEMBER_DELETION -> context.getString(R.string.fcm_notification_member_deletion, nickname)
+            NotificationType.ETA_NOTICE -> context.getString(R.string.fcm_notification_eta_notice, meetingName)
+            NotificationType.DEFAULT -> ""
+        }
 
-            NotificationType.NUDGE -> {
-                context.getString(R.string.fcm_notification_nudge, nickname)
+    private fun getPendingIntent(
+        type: NotificationType,
+        meetingId: String,
+    ): PendingIntent? {
+        val navigationTarget =
+            when (type) {
+                NotificationType.ENTRY,
+                NotificationType.DEPARTURE_REMINDER,
+                NotificationType.MEMBER_DELETION,
+                -> NAVIGATE_TO_NOTIFICATION_LOG
+                NotificationType.NUDGE,
+                NotificationType.ETA_NOTICE,
+                -> NAVIGATE_TO_ETA_DASHBOARD
+                NotificationType.DEFAULT -> null
             }
 
-            NotificationType.DEFAULT -> {
-                ""
-            }
+        return navigationTarget?.let { target ->
+            val intent = MeetingRoomActivity.getIntent(context, meetingId.toLong(), target)
+            PendingIntent.getActivity(
+                context,
+                NOTIFICATION_REQUEST_CODE,
+                intent,
+                PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT,
+            )
```

Comment:
```suggestion
        val intent = MeetingRoomActivity.getIntent(context, meetingId.toLong(), navigationTarget)
        return PendingIntent.getActivity(
                context,
                NOTIFICATION_REQUEST_CODE,
                intent,
                PendingIntent.FLAG_IMMUTABLE or PendingIntent.FLAG_UPDATE_CURRENT,
            )
```
let 때문에 이 함수가 어떤 걸 반환하는지 가독성이 떨어지는 것 같은데 이런 식으로 바꾸는 건 어떤가요? 개인적 의견입니다!

Reply:
좋습니다! [cc4d9c4](https://github.com/woowacourse-teams/2024-ody/pull/572/commits/cc4d9c412772a5ba860679bcb4288a810070e462)